aws/resource_aws_dx_hosted_private_virtual_interface_test.go

```diff
@@ -74,42 +157,62 @@ func testAccCheckAwsDxHostedPrivateVirtualInterfaceDestroy(s *terraform.State) e
 		if rs.Type != "aws_dx_hosted_private_virtual_interface" {
 			continue
 		}
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No ID is set")
+		}
 
-		input := &directconnect.DescribeVirtualInterfacesInput{
+		resp, err := conn.DescribeVirtualInterfaces(&directconnect.DescribeVirtualInterfacesInput{
 			VirtualInterfaceId: aws.String(rs.Primary.ID),
+		})
+		if isAWSErr(err, directconnect.ErrCodeClientException, "does not exist") {
+			continue
 		}
-
-		resp, err := conn.DescribeVirtualInterfaces(input)
 		if err != nil {
 			return err
 		}
-		for _, v := range resp.VirtualInterfaces {
-			if *v.VirtualInterfaceId == rs.Primary.ID && !(*v.VirtualInterfaceState == directconnect.VirtualInterfaceStateDeleted) {
-				return fmt.Errorf("[DESTROY ERROR] Dx Private VIF (%s) not deleted", rs.Primary.ID)
+
+		n := len(resp.VirtualInterfaces)
+		switch n {
+		case 0:
+			continue
+		case 1:
+			if aws.StringValue(resp.VirtualInterfaces[0].VirtualInterfaceState) == directconnect.VirtualInterfaceStateDeleted {
+				continue
 			}
+			return fmt.Errorf("still exist.")
+		default:
+			return fmt.Errorf("Found %d Direct Connect virtual interfaces for %s, expected 1", n, rs.Primary.ID)
```

Comment:
This feels more along the lines of testing the API response is valid versus the previous `for` loop method, which only worries about the matching result if it found. If the API begins returning too much information such as other VIFs, but does contain our result VIF, that seems to be what we should be more concerned with. 👍

Reply:
OK, I'll change back to the prior logic and retest.

Reply:
OK, I'll change back to the prior logic and retest.